modular_sand/code/datums/interactions/lewd_definitions.dm

```diff
@@ -787,6 +787,17 @@
 				H.mob_climax(TRUE, "masturbation", "none")
 			else
 				H.mob_climax(TRUE, "sex", partner, !cumin, target_gen)
+
+	// SPLURT edit
+	if(iscarbon(src))
+		var/mob/living/carbon/c_src
+
+		c_src = src
+		for(var/obj/item/organ/genital/G in c_src.internal_organs)
+			if(check_orgasm(G))
+				break
```

Comment:
I really hate this loop- it ostensibly doesn't do anything- In reality it's there for the side effects of check_orgasm. I'm not sure what a better way to do this would be right now.

Reply:
Hrm.. I can't think of a pretty way of doing that without a genital being used in the proc, please tell me if you have any ideas!